src/unix_socket.rs

```diff
@@ -0,0 +1,104 @@
+use anyhow::Result;
+use reqwest::blocking::{Request, Response};
+use reqwest::header::{HeaderValue, HOST};
+use std::path::PathBuf;
+use std::time::Instant;
+
+use crate::middleware::{Context, Middleware, ResponseMeta};
+use crate::utils::test_mode;
+
+pub struct UnixSocket {
+    rt: tokio::runtime::Runtime,
+    socket_path: PathBuf,
+}
+
+impl UnixSocket {
+    pub fn new(socket_path: PathBuf) -> Self {
+        let rt = tokio::runtime::Builder::new_multi_thread()
+            .enable_all()
+            .build()
+            .unwrap();
+
+        Self { rt, socket_path }
+    }
+
+    pub fn execute(&self, request: Request) -> Result<Response> {
+        self.rt.block_on(async {
+            // TODO: Support named pipes by replacing tokio::net::UnixStream::connect(..) with:
```

Comment:
Besides Docker for Window, I couldn't find any other examples of HTTP over named pipes, so I'm not sure if it's worth supporting at this time.